src/core/config/styles.js

```diff
@@ -32,23 +35,21 @@ function getCardBackground(isContrastMode = false) {
 
 export function getCurrentTheme(isContrastMode) {
     const backgroundImage = get(themeImages, 'backgroundImage', null);
-    const cardBackground = getCardBackground(isContrastMode);
-
+    const themeColor = (window && window.config) ? Object.assign({}, defaultTheme, window.config.theme) : defaultTheme;
```

Comment:
Why do you create an intermediate object? Merge the theme from config with the theme you're building in the following lines. That way, the theme from config can override all the settings you define in this theme. Something like:

```js
const theme = createMuiTheme(deepMerge(defaultTheme, window.config.theme));
``` 

That means the theme from the window.config will have to use the same structure as the mui theme, and store the colors under the palette key.

Reply:
@fzaninotto 
If I understand you correctly, I should do smth like this:
```
// index.html
window.config = {
            domainName: "https://192.168.88.38",
	    palette: {
                  mainColor: "#3596f4",
                  dangerColor: "#da534f",
                  contrastColor: "#000",
	     },
        }

// src/core/config/styles.js
const defaultTheme = {
     palette: {},
     ....
}

return createMuiTheme(deepmerge(defaultTheme,  window.config));
```

I tried to do this, but in this case we have three problems:

1) We use two types of palettes depends on isContrastMode. If we will use deepmerge, window.config will override this one:
```
palette: isContrastMode ? darkPalette : lightPalette,
```

2) We need in theme main color for some places, where we use image as background. If image is absent, we use main theme color:

```
const cardBackground = getCardBackground(isContrastMode, themeColor);
return createMuiTheme({
    tableHeader: {
            tableHeaderBlock: {
                background: cardBackground,
            },
        },
});
```
Or we should add this parameter to window.config.

3) We need in theme main color from window.config in UserTour component (I described this issue below).

Reply:
1. Store two palettes in the config

```js
window.config = {
            domainName: "https://192.168.88.38",
	    lightPalette: {
                  mainColor: "#3596f4",
                  dangerColor: "#da534f",
                  contrastColor: "#000",
	     },
            darkPalette: {
                  ...
            }
        }
```

2. I'm not sure I understand your question. If you need the theme color, use the function form of withStyles()

```js
const styles = theme =>> ({
    tableHeader: {
        background: theme.palette.main
    }
});
```

3. I'll answer below

Reply:
Added both palettes to window.config. They are merged with default theme in createMuiTheme: 
https://github.com/PulseTile/PulseTile-RA/blob/feature-optimization/src/core/config/styles.js